src/daft-local-execution/src/lib.rs

```diff
@@ -63,14 +66,31 @@ pub enum Error {
     OneShotRecvError {
         source: tokio::sync::oneshot::error::RecvError,
     },
+    #[snafu(display("Error creating pipeline from {}: {}", plan_name, source))]
+    PipelineCreationError {
+        source: DaftError,
+        plan_name: String,
+    },
+    #[snafu(display("Error when running pipeline node {}: {}", node_name, source))]
+    PipelineExecutionError {
+        source: DaftError,
+        node_name: String,
+    },
 }
 
 impl From<Error> for DaftError {
     fn from(err: Error) -> DaftError {
-        DaftError::External(err.into())
+        match err {
+            Error::PipelineCreationError { .. } | Error::PipelineExecutionError { .. } => {
+                DaftError::InternalError(err.to_string())
+            }
+            _ => DaftError::External(err.into()),
```

Comment:
We probably don't want to nest the execution error under `DaftError::External` otherwise the conversion of the DaftError to the correct PythonError [won't happen](
https://github.com/Eventual-Inc/Daft/blob/b15b0c7f7189e4ec395cb9d0491854ceae1bfda5/src/common/error/src/python.rs#L19).

For now, we can unnest the daft error, log where it came from and then return that.

Reply:
Done and updated PR description example. Assumed you meant not to nest DaftErrors under `DaftError::InternalError`, the Externals are for the tokio oneshot and join errors.

Reply:
oh whoops, yeah looks good now!